File: platform/polyglot/app_yaml_static_config/py/src/app_yaml_static_config/sdk.py
from typing import Any, Dict, List, Optional
import copy
import glob
import os
from .core import AppYamlConfig
from .types import InitOptions

class AppYamlConfigSDK:
    def __init__(self, config: AppYamlConfig):
        self.config = config

    @classmethod
    def from_directory(cls, config_dir: str) -> 'AppYamlConfigSDK':
        yaml_files = glob.glob(os.path.join(config_dir, "*.yaml"))
        yml_files = glob.glob(os.path.join(config_dir, "*.yml"))
        files = yaml_files + yml_files
        AppYamlConfig.initialize(InitOptions(files=files, config_dir=config_dir))
        return cls(AppYamlConfig.get_instance())

    def get(self, key: str) -> Any:
        value = self.config.get(key)
        return copy.deepcopy(value)

    def get_nested(self, keys: List[str]) -> Any:
        value = self.config.get_nested(*keys)
        return copy.deepcopy(value)

    def get_all(self) -> Dict[str, Any]:
        return copy.deepcopy(self.config.get_all())

    def list_providers(self) -> List[str]:
        providers = self.config.get('providers', {})
        return list(providers.keys())

    def list_services(self) -> List[str]:
        services = self.config.get('services', {})
        return list(services.keys())

    def list_storages(self) -> List[str]:
        storage = self.config.get('storage', {})
        return list(storage.keys())

File: platform/polyglot/app_yaml_static_config/mjs/src/sdk.ts
import { glob } from 'glob';
import * as path from 'path';
import { AppYamlConfig } from './core.js';
import { InitOptions } from './types.js';

export class AppYamlConfigSDK {
    private config: AppYamlConfig;

    constructor(config: AppYamlConfig) {
        this.config = config;
    }

    static async fromDirectory(configDir: string): Promise<AppYamlConfigSDK> {
        const files = await glob(path.join(configDir, "*.{yaml,yml}"));
        await AppYamlConfig.initialize({ files, configDir });
        return new AppYamlConfigSDK(AppYamlConfig.getInstance());
    }

    get(key: string): unknown {
        const value = this.config.get(key);
        return value === undefined ? undefined : structuredClone(value);
    }

    getNested(keys: string[]): unknown {
        const value = this.config.getNested(keys);
        return value === undefined ? undefined : structuredClone(value);
    }

    getAll(): Record<string, unknown> {
        return structuredClone(this.config.getAll());
    }

    listProviders(): string[] {
        const providers = this.config.get<Record<string, unknown>>('providers') ?? {};
        return Object.keys(providers);
    }

    listServices(): string[] {
        const services = this.config.get<Record<string, unknown>>('services') ?? {};
        return Object.keys(services);
    }

    listStorages(): string[] {
        const storage = this.config.get<Record<string, unknown>>('storage') ?? {};
        return Object.keys(storage);
    }
}
